pkg/aws/metadata/server.go

```diff
@@ -34,16 +36,18 @@ type Server struct {
 }
 
 type ServerConfig struct {
-	ListenPort       int
-	MetadataEndpoint string
-	AllowIPQuery     bool
+	ListenPort           int
+	MetadataEndpoint     string
+	AllowIPQuery         bool
+	WhitelistRouteRegexp *regexp.Regexp
 }
 
-func NewConfig(port int) *ServerConfig {
+func NewConfig(port int, allowIPQuery bool, whitelistRouteRegexp *regexp.Regexp) *ServerConfig {
```

Comment:
I'd rather we didn't expand NewConfig to include all options.

To be honest, I'd prefer it didn't specify a port but set good defaults for everything and then clients set individual fields.

Otherwise, as the struct grows we're just adding more and more params to the function which isn't nice.

Reply:
Maybe we should rename the func to `DefaultConfig` too.